src/utils/recorder_events.jl

```diff
@@ -81,58 +81,59 @@ struct ParameterUpdateEvent <: IS.AbstractRecorderEvent
     parameter_type::String
     device_type::String
     tag::String
-    problem_number::Int
+    model_name::String
 end
 
 function ParameterUpdateEvent(
     parameter_type::Type{<:ParameterType},
     device_type::Type{<:PSY.Device},
     tag::String,
     execution_timestamp::Dates.DateTime,
-    problem_number::Int,
+    model_name,
 )
     return ParameterUpdateEvent(
         IS.RecorderEventCommon("ParameterUpdateEvent"),
         execution_timestamp,
         string(parameter_type),
         string(device_type),
         tag,
-        problem_number,
+        string(model_name),
     )
 end
 
-struct FeedforwardUpdateEvent <: IS.AbstractRecorderEvent
+struct StateUpdateEvent <: IS.AbstractRecorderEvent
     common::IS.RecorderEventCommon
-    category::String
-    simulation_time::Dates.DateTime
-    parameter_type::String
-    device_name::String
-    previous_value::Float64
-    val::Float64
-    problem_number::Int
-    source::Int
+    execution_timestamp::Dates.DateTime
+    entry_type::String
+    component_type::String
+    model_name::String
 end
 
-function FeedforwardUpdateEvent(
-    category::String,
-    simulation_time::Dates.DateTime,
-    parameter::VariableValueParameter,
-    device_name::String,
-    val::Float64,
-    previous_value::Float64,
-    destination_model::DecisionModel,
-    source_model::DecisionModel,
+function StateUpdateEvent(
+    entry_type::DataType,
+    component_type::DataType,
+    execution_timestamp::Dates.DateTime,
+    model_name,
 )
-    return FeedforwardUpdateEvent(
-        IS.RecorderEventCommon("FeedforwardUpdateEvent"),
-        category,
-        simulation_time,
-        parameter,
-        device_name,
-        previous_value,
-        val,
-        get_simulation_number(destination_problem),
-        get_simulation_number(source_problem),
+    return StateUpdateEvent(
+        IS.RecorderEventCommon("StateUpdateEvent"),
+        execution_timestamp,
+        string(entry_type),
+        string(component_type),
+        string(model_name),
+    )
+end
+
+function StateUpdateEvent(
```

Comment:
Should we make a script that automatically generates files with textual tables of all recorder events? That could make it very easy to browse these. I could take care of it later.

Reply:
That's a good idea, we should do this.